packages/playwright-core/src/client/page.ts

```diff
@@ -476,6 +483,36 @@ export class Page extends ChannelOwner<channels.PageChannel> implements api.Page
     return buffer;
   }
 
+  async _expectScreenshot(options: ExpectScreenshotOptions): Promise<{ actual?: Buffer, expected?: Buffer, diff?: Buffer, errorMessage?: string , log?: string[]}> {
+    const mask = options.screenshotOptions?.mask ? options.screenshotOptions?.mask.map(locator => ({
+      frame: locator._frame._channel,
+      selector: locator._selector,
+    })) : undefined;
+    const locator = options.locator ? {
+      frame: options.locator._frame._channel,
+      selector: options.locator._selector,
+    } : undefined;
+    const expected = options.expected ? options.expected.toString('base64') : undefined;
+
+    const result = await this._channel.expectScreenshot({
+      ...options,
+      isNot: !!options.isNot,
+      expected,
+      locator,
+      screenshotOptions: {
+        ...(options.screenshotOptions || {}),
```

Comment:
```suggestion
        ...options.screenshotOptions,
```
nit: spread operator should work with null or undefined.